### ISSUE TITLE
Can't block http and https traffic except for a website

### ISSUE THREAD

**Issue by dehbini** [hwdsl2/setup-ipsec-vpn#207]
Hi
I really appreciate for your nice script.I really apologize for this simple question because I'm new to this field.I want to block all http and https traffic except for websites that have ip range on port 443 (XXX.0.0.0/8) . So clients can just access and open websites that are in this range ( XXX.0.0.0/8) when they are connected to VPN.I have added these rules after **"-A INPUT -p udp --dport 1701 -j DROP" in /etc/iptables.rules** 
```
-A INPUT -p tcp -m multiport --dports 443,80 -j DROP
-A INPUT -p tcp -d XXX.0.0.0/8 --dport 443 -j ACCEPT
```
and then reboot the server. but when I connect to VPN, I can access all websites and can't block http and https traffic. Would u plz help me solve this problem?

**Comment by hwdsl2** [comment]
@dehbini Regarding your question: Instead of IPTables INPUT rules, use FORWARD rules (example below). Put them on top of the FORWARD chain in `/etc/iptables.rules`. Then reboot your server.

```
-A FORWARD -i ppp+ -o eth+ -p tcp -d XXX.0.0.0/8 --dport 443 -j ACCEPT
-A FORWARD -i ppp+ -o eth+ -p tcp -m multiport --dports 443,80 -j DROP
-A FORWARD -s 192.168.43.0/24 -o eth+ -p tcp -d XXX.0.0.0/8 --dport 443 -j ACCEPT
-A FORWARD -s 192.168.43.0/24 -o eth+ -p tcp -m multiport --dports 443,80 -j DROP
```

**Comment by dehbini** [comment]
@hwdsl2 I have edited the IPTables as you say and reboot there server but when I connect to VPN it opens all http and https website again.This is my new FORWARD chain
```
-A FORWARD -i ppp+ -o eth+ -p tcp -d 216.0.0.0/8 --dport 443 -j ACCEPT
-A FORWARD -i ppp+ -o eth+ -p tcp -m multiport --dports 443,80 -j DROP
-A FORWARD -s 192.168.43.0/24 -o eth+ -p tcp -d 216.0.0.0/8 --dport 443 -j ACCEPT
-A FORWARD -s 192.168.43.0/24 -o eth+ -p tcp -m multiport --dports 443,80 -j DROP
-A FORWARD -m conntrack --ctstate INVALID -j DROP
-A FORWARD -i ens32 -o ppp+ -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -i ppp+ -o ens32 -j ACCEPT
-A FORWARD -s 192.168.42.0/24 -d 192.168.42.0/24 -i ppp+ -o ppp+ -j ACCEPT
-A FORWARD -d 192.168.43.0/24 -i ens32 -m conntrack --ctstate RELATED,ESTABLISHED -j ACCEPT
-A FORWARD -s 192.168.43.0/24 -o ens32 -j ACCEPT
-A FORWARD -j DROP
```

**Comment by hwdsl2** [comment]
@dehbini Replace all "eth+" with "ens32" in the four new rules in your
FORWARD chain, save the file and reboot. This should fix it.

**Comment by dehbini** [comment]
@hwdsl2 Thanks. I did it and this solved the problem but it blocks all http and https traffic and can't access google.com or 216.58.206.110 as they are accepted in these rules
```
-A FORWARD -i ppp+ -o ens32 -p tcp -d 216.0.0.0/8 --dport 443 -j ACCEPT
-A FORWARD -i ppp+ -o ens32 -p tcp -m multiport --dports 443,80 -j DROP
-A FORWARD -s 192.168.43.0/24 -o ens32 -p tcp -d 216.0.0.0/8 --dport 443 -j ACCEPT
-A FORWARD -s 192.168.43.0/24 -o ens32 -p tcp -m multiport --dports 443,80 -j DROP
```

**Comment by hwdsl2** [comment]
@dehbini You can't achieve this with IPTables because google.com could
return different IPs in many different subnets for its A record, not just
216.x.x.x. If you visit that IP directly in your VPN client it should
connect.

**Comment by dehbini** [comment]
@hwdsl2 Thanks, you are right. Is there any method, to access only google.com? is it possible to use the domain name except IP?

**Comment by hwdsl2** [comment]
@dehbini You can try that but I doubt it will work.

**Comment by dehbini** [comment]
@hwdsl2 Hi again. I have installed the latest script recently and I see that IPtables has been changed. You have add :ufw-xxx . I'm now confused how to edit rules to achieve my goal like before. "I want to block all http and https traffic except for websites that have ip range on port 443 and 80 (XXX.0.0.0/8) . So clients can just access and open websites that are in this range ( XXX.0.0.0/8) when they are connected to VPN." I really appreciate your time and effort solving my issue.
